src/clustering.jl

```diff
@@ -34,21 +34,21 @@ function cluster_loads!(grid::Grid, nload)
     end
     sum_pops = Symmetric(sum_pops, :L)
     sub_dist = Matrix(load_sub_dist(substations(grid), load_dist, id2ind, sum_pops))
-    const POPLIMIT = 200000 # population limit for a substation (=> 400MW load)
+    POPLIMIT = 200000 # population limit for a substation (=> 400MW load)
 
     while length(substations(grid)) > nload
         n = length(substations(grid))
         ok = false
         i = j = -1
         while true
-            m = indmin(sub_dist)
+            m = findmin(sub_dist)[2]  # [2] is the index of the smallest item
             if sub_dist[m] == Inf # Can't cluster anymore without going over POPLIMIT
                 warn("Can't cluster anymore without going over population limit!")
                 warn("Stopping clustering process before target number is reached")
                 nload = Inf
                 break
             end
-            i, j = ind2sub(size(sub_dist), m)
+            i, j = Tuple(CartesianIndices(size(sub_dist))[m])
```

Comment:
What is this line doing? I don't understand what it was doing before (I thought `ind2sub` was a `Dict`, but `Dict`s are not callable like `ind2sub(x, y)`), but my worry is this has different behaviour...

Reply:
It takes a linear index and turns into the 2D index of a matrix. e.g.: If one has a 2x4 matrix, it maps 5 to (2, 1).

Reply:
well, it does select the index, but it iterates column-wise (not row-wise) -- just checking this is the behaviour you want
```julia
julia> X = rand(2, 4);

julia> CartesianIndices(size(X))
2×4 CartesianIndices{2,Tuple{Base.OneTo{Int64},Base.OneTo{Int64}}}:
 CartesianIndex(1, 1)  CartesianIndex(1, 2)  CartesianIndex(1, 3)  CartesianIndex(1, 4)
 CartesianIndex(2, 1)  CartesianIndex(2, 2)  CartesianIndex(2, 3)  CartesianIndex(2, 4)

julia> CartesianIndices(size(X))[5]
CartesianIndex(1, 3)
```

Reply:
Botched example of mine. Good thing I was not the one who implemented `CartesianIndices` =P

Reply:
Can't this be simplified to:
```julia
i, j = Tuple(m)
```
It used to be that `indmin` returned a linear index but now both `findmin` and `argmin` return `CartesianIndex`